<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
<link rel="stylesheet" type="text/css" href="http://forest.mbl.edu/gbrowse/main.css">
<html lang="en">

<head><title>JBPC Generic Model Organism Database Servers</title></head>
<body bgcolor="white" link="#2F588F" vlink="#2F588F" alink="#AFAE6B">

<h1 align="CENTER">JBPC Generic Model Organism Database Servers</h1>
<p><center>(<a href="http://jbpc.mbl.edu/computational-resources.html">return to the JBPC Computing home page</a>)</center></p>
<p><center><em><font color=blue>Access to some GMOD/GBrowse resources and databases is firewall or password protected - contact <a href="mailto:biocomp@lists.mbl.edu">biocomp@lists.mbl.edu</a> for details.</font></em></center></p>

<hr>
<font size=2><!-- include this into .shtml files -->

	<p><b>GMOD.MBL.EDU</b> is our cluster of Advanced Genome Browsers, used for creating online resources for combining and analyzing genome, gene expression, and annotation data for genome-scale sequence data. It is the primary bioinformatics engine for the third phase of the <a href="http://www.mbl.edu/Giardia">GiardiaDB</a> project, by incorporating genome assembly, genome annotation, gene expression, and high-throughput phylogenetic information into the <a href="http://www.gmod.org">Generic Model Organism Database</a> (GMOD) paradigm. The GMOD server can be used to support any ongoing genome-level analysis in the JBPC and is currently being used for <em>Antonospora</em>, <em>Blochmania</em>, <em>Trypanosoma</em>, EST, rotifer, and many other projects.</p>
	<p><b>Questions, requests, training help, and bug reports should be emailed to <a href="mailto:biocomp@lists.mbl.edu">biocomp@lists.mbl.edu</a>.</b></p>
	<hr>
	<h2>Features</h2>
	<p>Add your own genome sequencing data to GMOD. GMOD can support Arachne, TIGR, and PHRAP assemblies and CONSED assembly editing results as part of our collaboration with the <a href="http://www.tigr.org/software/AMOS/">AMOS Project</a>.  We have a suite of assembly analysis tools allowing assessment of read coverage, scaffolding, and closure predictions.</p>
	<p>Our GMOD servers can integrate <a href="http://www.sagenet.org">Serial Analysis of Gene Expression</a> (SAGE) and <a href="http://www.lynxgen.com/wt/tert.php3?page_name=mpss">Massively Parallel Signature Sequencing</a> (MPSS) results with partial genomes, complete genomes, or EST sequences.  Advanced gene expression analysis tools are included.</p>
	<p>Your annotations and genome features (ORFs, ESTs, tRNAs, repeats, etc) can be mapped to your genome sequences.  The <a href="http://www.sanger.ac.uk/Software/formats/GFF/GFF_Spec.shtml">General Feature Format</a> is preferred, as are the <a href="http://www.geneontology.org/GO.evidence.shtml">Gene Ontology Evidence Codes</a>.  Alternatively, you can use our genome annotation pipeline to predict protein and functional RNA encoding genes, repeats, assembly mapping, plus functional annotation of predicted proteins (InterPro domains, signal peptides, transmembrane domains, etc.).<p>
	<p>Perform complex queries by direct access to your data using SQL.  Suggested SQL clients are <a href="http://www.dbtools.com.br/EN/">DBTools</a> (Windows) and <a href="http://code.google.com/p/sequel-pro/">sequel-pro</a> (Mac OSX).</p>
	<p><a href="http://www.gmod.org/ggb/index.shtml">GBrowse</a> is the advanced web-based graphical genome browser used by our GMOD projects.  You can easily customize the appearance of your data and perform a number of types of searches from within GBrowse.  To get started, view the <a href="http://forest.mbl.edu/cgi-bin/site/giardia?page=general_help">GBrowse General Help</a> (GiardiaDB version).</p>
	<p>For a pdf describing all of the various features of the databases, click <a href="/gmoddb_overview.pdf">here</a></p>
	<p>For a pdf detailing how to perform cluster analysis of SAGE data, click <a href="/SAGE-ClusterAnalysis.pdf">here</a></p>
	<p>For a pdf detailing raw SAGE data processing in preparation for import to GMOD, click <a href="/General-SAGE-Processing.pdf">here</a></p>
	<hr>
	<h2>JBPC Protist EST Projects</h2>
	<p>A large number of protist EST sequencing projects are underway at the JBPC.  Currently, each set of EST data can be accessed separately but expect a meta-server for comparing shared EST-encoded homologs among organisms soon.</p>
	<p>The EST data are currently <b>Private</b>.  The lead investigator is <a href="mailto:morrison@mbl.edu">Hilary Morrison</a>.  Please email any questions about these data to <a href="mailto:biocomp@lists.mbl.edu">biocomp@lists.mbl.edu</a>.</p>
	<table border=0>
	<tr>
	<td width="50%">
		<ul>
		<li><a href="http://forest.mbl.edu/cgi-bin/site/acantha">Acanthamoeba astronyxis</a>
		<li><a href="http://forest.mbl.edu/cgi-bin/site/arachnu">Arachnula sp.</a>
		<li><a href="http://forest.mbl.edu/cgi-bin/site/bodoxxx">Bodo saltans</a>
		<li><a href="http://forest.mbl.edu/cgi-bin/site/chilomo">Chilomonas paramecium</a>
		<li><a href="http://forest.mbl.edu/cgi-bin/site/diplone">Diplonema papillatum</a>
		<li><a href="http://forest.mbl.edu/cgi-bin/site/euglena">Euglena mutabilis</a>
		<li><a href="http://forest.mbl.edu/cgi-bin/site/euplote">Euplotes crassus</a>
		</ul>
	</td>
	<td width="50%">
		<ul>
		<li><a href="http://forest.mbl.edu/cgi-bin/site/giardia_est">Giardia lamblia</a>
		<li><a href="http://forest.mbl.edu/cgi-bin/site/heterom">Heteromita sp.</a>
		<li><a href="http://forest.mbl.edu/cgi-bin/site/nannoch">Nannochloropsis granulata</a> 
		<li><a href="http://forest.mbl.edu/cgi-bin/site/pelagom">Pelagomonas calceolata</a> 
		<li><a href="http://forest.mbl.edu/cgi-bin/site/pinguic">Pinguiococcus pyrenoidosus</a>
		<li><a href="http://forest.mbl.edu/cgi-bin/site/tribone">Tribonema aequale</a>
		</ul>
	</td>
	</tr>
	</table>
	
	<br><hr>
	<h2>All Other Projects</h2>
	<table border=1>
	<tr>
		<td width="15%"><center><b>Project</b></center></td>
		<td><center><b>Description</b></center></td>
		<td width="10%"><center><b>Access</b></center></td>
		<td width="20%"><center><b>Investigator</b></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/antonospora01">AntonosporaDB</a></center><br></td>
		<td>Website for the <em>Antonospora locustae</em> genome project. </td>
		<td><center><font color="blue"><b>Public</font></b></center></td>
		<td><center><a href="mailto:morrison@mbl.edu">Hilary Morrison</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/aureococcus">Aureococcus</a></center><br></td>
		<td>The <em>Aureococcus</em> Serial Analysis of Gene Expression (SAGE) project.  Genome and EST data acquired from GenBank and other external sources.</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:sdyhrman@whoi.edu">Sonya Dyhrman</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/bpennsylvanicus">Blochmannia pennsylvanicus</a></center><br></td>
		<td>The genome sequence of <em>Blochmannia pennsylvanicus</em> (GenBank CP000016), with addition of Serial Analysis of Gene Expression data.</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:jwernegreen@mbl.edu">Jennifer Wernegreen</a>, <a href="mailto:amcarthur@mac.com">Andrew McArthur</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/bmanjavacas4">Brachionus manjavacas cDNA</a></center><br></td>
		<td>The <em>Brachionus manjavacas</em> RUS cDNA library assembly and annotation database (version 4).</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:dmarkwelch@mbl.edu">David Mark Welch</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/bmanjavacasEST">Brachionus manjavacas EST</a></center><br></td>
		<td>The <em>Brachionus manjavacas EST</em> BLASTable database.  For comparative blasting purposes only, the GBrowse assembly is not currently available.</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:dmarkwelch@mbl.edu">David Mark Welch</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/bplicatilisa">Brachionus plicatilis mixed EST</a></center><br></td>
		<td>The <em>Brachionus plicatilis</em> NH1L mixed population, unnormalized EST library assembly and annotation database described in Suga <i>et al.</i> 2007. <i>PLoS ONE<i> <a href="http://www.plosone.org/article/fetchArticle.action?articleURI=info%3Adoi%2F10.1371%2Fjournal.pone.0000671">2(8): e671. doi:10.1371/journal.pone.0000671.</a></td>
		<td><center><font color="blue"><b>Public</font></b></center></td>
		<td><center><a href="mailto:dmarkwelch@mbl.edu">David Mark Welch</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/bplicatilis_male">Brachionus plicatilis male EST</a></center><br></td>
		<td>The <em>Brachionus plicatilis</em> NH1L male, normalized EST library assembly and annotation database.</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:dmarkwelch@mbl.edu">David Mark Welch</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/bplicatilis_amictic">Brachionus plicatilis amictic EST</a></center><br></td>
		<td>The <em>Brachionus plicatilis</em> NH1L amictic female, normalized EST library assembly and annotation database.</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:dmarkwelch@mbl.edu">David Mark Welch</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/brachionus">Combined Brachionus sp cDNA</a></center><br></td>
		<td>The <em>Combined Brachionus sp.</em> cDNA library assembly and annotation database including data from B.plicatilis and B.manjavacas.</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:dmarkwelch@mbl.edu">David Mark Welch</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/emiliania04">Emiliania huxleyi</a></center><br></td>
		<td>The <em>Emiliania huxleyi</em> Serial Analysis of Gene Expression (SAGE) project.  Genome and EST data acquired from GenBank and other external sources.</td>
		<td><center><font color="blue"><b>Public</font></b></center></td>
		<td><center><a href="mailto:sdyhrman@whoi.edu">Sonya Dyhrman</a>, <a href="mailto:amcarthur@mac.com">Andrew McArthur</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/ebieneusi">Enterocytozoon bieneusi</a></center><br></td>
		<td>The <em>Enterocytozoon ebieneusi</em> Whole genome shotgun sequencing of E. bieneusi</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:morrison@mbl.edu">Hilary Morrison</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/env396_fosmids_6_56">Environmental Sampling Site 396, Fosmids_6_ 56</a></center><br></td>
		<td><em>Fosmids 6 and 56</em> are genetic sequences from unknown organisms collected at the hydrothermal vent sampling site 396</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:jhuber@mbl.edu">Julie Huber</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/env396_fosmid68">Environmental Sampling Site 396, Fosmid 68</a></center><br></td>
		<td><em>Fosmid 68</em> is a genetic sequence from an unknown organism collected at the hydrothermal vent sampling site 396</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:jhuber@mbl.edu">Julie Huber</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/env396_fosmid87">Environmental Sampling Site 396, Fosmid 87</a></center><br></td>
		<td><em>Fosmid 87</em> is a genetic sequence from an unknown organism collected at the hydrothermal vent sampling site 396</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:jhuber@mbl.edu">Julie Huber</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/fos20J22">Fosmid 20J22</a></center><br></td>
		<td><em>Fosmid 20J22</em> assembly of a test fosmid using capillary, GS20 and FLX data</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:dmarkwelch@mbl.edu">David Mark Welch</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/giardia14">GiardiaDB</a><br>(internal use only)</center><br></td>
		<td>Internal copy of the <em>Giardia lamblia</em> genome project for JBPC use only.<br><b>The public version of <a href="http://www.giardiadb.org">GiardiaDB</a> has moved to <a href="http://www.giardiadb.org">www.giardiadb.org</a></b> <br>at the National Institute of Allergy and Infectious Diseases <a href="http://www.niaid.nih.gov/dmid/genomes/brc/default.htm">Bioinformatics Resource Center</a></td>
		<td><center>Private</center></td>
		<td><center><a href="http://jbpc.mbl.edu/contact.php?recipient_code=200">Giardia Contact Form</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/marinobacter_jgi">Marinobacter aquaeolei</a><br>(complete genome)</center><br></td>
		<td>Annotation of the complete <em>Marinobacter aquaeolei</em> genome, using genome data acquired from the DOE Joint Genome Institute.</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:shuse@mbl.edu">Susan Huse</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/proseola">Philodina roseola</a></center><br></td>
		<td>Sequence assembly and automated annotation of cosmid and other sequences of the rotifer <em>Philodina roseola</em>.</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:dmarkwelch@mbl.edu">David Mark Welch</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/proseolacdna">Philodina roseola cDNA</a></center><br></td>
		<td>Normalized cDNA library for <em>Philodina roseola</em> (PrA library).</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:dmarkwelch@mbl.edu">David Mark Welch</a></center></td>
	</tr>

	<tr>
		<td><br><center><a href="http://forest.mbl.edu/cgi-bin/site/s_mansoniest">Schistosoma mansoni EST</a></center><br></td>
		<td>Annotation of <em>Schistosoma mansoni</em> EST data, including mapping of results of Serial Analysis of Gene Expression (SAGE) experiments.  EST sequence data acquired from the TIGR and ONSA genome centers.</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:amcarthur@mac.com">Andrew McArthur</a></center></td>
	</tr>

	<tr>
		<td>
			<br><center><i>Synechococcus</i><br><a href="http://forest.mbl.edu/cgi-bin/site/syn_4320_2">4320_2</a>, <a href="http://forest.mbl.edu/cgi-bin/site/syn_rs9916">RS9916</a>, <a href="http://forest.mbl.edu/cgi-bin/site/syn_cc9605">CC9605</a>, <a href="http://forest.mbl.edu/cgi-bin/site/syn_cc9902">CC9902</a>, <a href="http://forest.mbl.edu/cgi-bin/site/syn_rs9917">RS9917</a>, <a href="http://forest.mbl.edu/cgi-bin/site/syn_wh5701">WH5701</a>, <a href="http://forest.mbl.edu/cgi-bin/site/syn_wh7805">WH7805</a></center><br>
		</td>
		<td>Annotation of <em>Synechococcus</em> genome assemblies.</td>
		<td><center>Private</center></td>
		<td><center><a href="mailto:anton.post@huji.ac.il">Anton Post</a></center></td>
	</tr>

	</table>

</font>
</body>
</html>
